File: frontend/types/similar_incident.ts
export interface SimilarIncidentResult {
  incident_id: string;
  incident_type: string;
  severity: string;
  junction_id: string;
  junction_name: string;
  timestamp: string;
  weather: string;
  similarity_score: number;
  matched_factors: string[];
  weak_match: boolean;
}

export interface SimilarIncidentsResponse {
  query_incident_id: string;
  results: SimilarIncidentResult[];
}

File: backend/app/schemas/similar_incident.py
from __future__ import annotations

from datetime import datetime
from pydantic import BaseModel


class SimilarIncidentResult(BaseModel):
    incident_id: str
    incident_type: str
    severity: str
    junction_id: str
    junction_name: str
    timestamp: datetime
    weather: str
    similarity_score: int
    matched_factors: list[str]
    weak_match: bool


class SimilarIncidentsResponse(BaseModel):
    query_incident_id: str
    results: list[SimilarIncidentResult]
